salt/client/ssh/__init__.py

```diff
@@ -589,11 +597,40 @@ def handle_ssh(self, mine=False):
                 log.error("No matching targets found in roster.")
                 break
             if len(running) < self.opts.get("ssh_max_procs", 25) and not init:
-                try:
-                    host = next(target_iter)
-                except StopIteration:
+                if targets_queue:
+                    host = targets_queue.pop(0)
```

Comment:
How large is `targets_queue`? `list.pop(0)` copies all pointers of the remaining elements, is that a problem here?

Reply:
Maximum size of the list is the number of salt-ssh systems to run the events on. In almost 99% it's just 1 element. In Uyuni there is a possibility to run salt-ssh event for multiple minions in one event is to use `Remote command` with wildcard, so the maximum number of the list in this case is the number of registered salt-ssh systems on the Server.

Reply:
Then I think this is fine, otherwise we could swap it out for a https://docs.python.org/3/library/collections.html#collections.deque, which actually is a linked list and not an array of pointers.

Reply:
@agraul could you please check https://github.com/openSUSE/salt/pull/493/commits/d2564f1954ad6ab253575390443c704e638ea51a

In stress test looks fine.